cumulusci/tasks/bulkdata.py

```diff
@@ -602,7 +678,62 @@ def _import_results(self, mapping, result_file, conn):
             process_incoming_rows(result_file, record_type), self.logger
         )
         data_file = IteratorBytesIO(processor)
-        self._sql_bulk_insert_from_csv(conn, mapping["table"], columns, data_file)
+        if mapping["oid_as_pk"]:
+            self._sql_bulk_insert_from_csv(conn, mapping["table"], columns, data_file)
+        else:
+            with tempfile.TemporaryFile("w+b") as f_values:
+                with tempfile.TemporaryFile("w+b") as f_ids:
+                    data_file_values, data_file_ids = self._split_batch_csv(
+                        data_file, f_values, f_ids
+                    )
+                    self._sql_bulk_insert_from_csv(
+                        conn, mapping["table"], columns, data_file_values
+                    )
+                    self._sql_bulk_insert_from_csv(
+                        conn, mapping["sf_id_table"], ["sf_id"], data_file_ids
+                    )
+
+        self.session.commit()
+
+        if lookup_keys and not mapping["oid_as_pk"]:
+            self._convert_lookups_to_id(mapping, lookup_keys)
+
+    def _get_mapping_for_table(self, table):
+        """ Returns the first mapping for a table name """
+        for mapping in self.mappings.values():
+            if mapping["table"] == table:
+                return mapping
+
+    def _split_batch_csv(self, data_file, f_values, f_ids):
+        writer_values = unicodecsv.writer(f_values)
+        writer_ids = unicodecsv.writer(f_ids)
+        for row in unicodecsv.reader(data_file):
+            writer_values.writerow(row[:-1])
+            writer_ids.writerow([row[-1]])
+        f_values.seek(0)
+        f_ids.seek(0)
+        return f_values, f_ids
+
+    def _convert_lookups_to_id(self, mapping, lookup_keys):
+        for lookup_key in lookup_keys:
+            lookup_dict = mapping["lookups"][lookup_key]
+            model = self.models[mapping["table"]]
+            lookup_mapping = self._get_mapping_for_table(lookup_dict["table"])
+            lookup_model = self.models[lookup_mapping["sf_id_table"]]
+            key_field = lookup_dict.get("key_field", lookup_key.lower())
+            key_attr = getattr(model, key_field)
+            try:
+                self.session.query(model).filter(
+                    key_attr.isnot(None), key_attr == lookup_model.sf_id
+                ).update({key_attr: lookup_model.id}, synchronize_session=False)
```

Comment:
I doubt this will perform well at large scale -- a big reason I switched to using oids as pks was so that we could simply insert (which is much faster especially using COPY FROM) without updating. So I'm glad you're still avoiding this when using oids as pks, but we should probably cover best practices for performance/scalability in docs.

Reply:
Added note to the docs about this.